chapter2.adoc

```diff
@@ -1,214 +1,207 @@
+This chapter introduces Sdsec ISA extension, which enhances the Sdext of _The RISC-V Debug Specification_ cite:[dbgspec]. The Sdsec extension provides privilege based protection for debug operations and trigger behaviors cite:[dbgspec]. Furthermore, it constrains trace functionality cite:[etrace] according to RISC-V privilege levels. 
 
+[[sdsecextdbg]]
 === External Debug
 
-By default, the extension forbids the following debug operation in <<dbops>> unless they are explicitly granted to the external debugger.
-	
+Chapter 3 of _The RISC-V Debug Specification_ cite:[dbgspec] outlines all mandatory and optional debug operations. The operations listed below are affected by the Sdsec extension, other operations remain unaffected. In the context of this chapter, *debug operations* refer to those listed below.
+
 [[dbops]]
-.The debug operations affected by Sdsec extension
-* Entering Debug Mode                                                     
+.Debug operations affected by Sdsec extension
+* Halting the hart to enter Debug Mode
 * Executing Program buffer                                                
 * Serving abstract commands (Access Register, Quick Access, Access Memory)
 
-When the debug operations are not granted, all the above operations issued by Debug Module or trigger will be pending or dropped. The subsequent subsections detail how debug operations are granted. 
-
-[[dbgaccpriv]]
-==== Debug Access Privilege
+When the external debug is disallowed in running privilege level, the hart behaves as the following: 
 
-The *debug access privilege* is defined as the privilege with which abstract commands or instructions in program buffers access hardware resources such as registers and memory. This privilege operates independently of hart privilege levels and exclusively affects operations within Debug Mode. Memory and register access within Debug Mode are subject to the *debug access privilege*, with all hardware protections, including MMU, PMP, and PMA, checked against it. This privilege is represented by the `prv` and `v` fields in `dcsr`, and it is updated to reflect the hart privilege level upon entering Debug Mode. Each hart has a dedicated *debug access privilege* and it may vary from each other. The permissible privilege levels programmable to `dcsr` in Debug Mode are elaborated in subsequent sections.
+[[dbgdisallowed]]
+* The hart will not enter Debug Dode on halt request (haltreq), single step or halt group. Halt request will be pending until debug is allowed again  
+* Triggers (with `action`=1) will not match or fire 
+* Abstract commands without halting (Access Register and Quick Access) will be dropped and set `cmderr` to 6 
```

Comment:
```suggestion
* Abstract commands without halting and Quick Access will be dropped and set `cmderr` to 6 
```
Rationale: 
- Quick Access halts and resume again (without halting is not true). 
- Abstract command without halting is more generic then access registers (it might be accessing memory)

Reply:
The following sections also use access register without halting instead of abstract commands without halting and Quick Access. Please more details in RISC-V Debug Spec 3.11.